asistente_ladm_col/utils/qgis_utils.py

```diff
@@ -1041,24 +1046,118 @@ def show_etl_model(self, db, input_layer, ladm_col_layer_name, geometry_type=Non
         if model:
             automatic_fields_definition = self.check_if_and_disable_automatic_fields(db, ladm_col_layer_name)
 
-            mapping = get_refactor_fields_mapping(ladm_col_layer_name, self)
+            if save_field_mapping is not None:
+                if self.func_load_field_mapping(ladm_col_layer_name, save_field_mapping) is not None:
+                    mapping = self.func_load_field_mapping(ladm_col_layer_name, save_field_mapping)
```

Comment:
Better to cache the result of the first call to `self.func_load_field_mapping(ladm_col_layer_name, save_field_mapping)` and then use such value everywhere.